website/content/en/docs/kubebuilder/references/logging.md

```diff
@@ -0,0 +1,261 @@
+
+
+Operator SDK-generated operators use the [`logr`][godoc_logr] interface to log. This log interface has several backends such as [`zap`][repo_zapr], which the SDK uses in generated code by default. [`logr.Logger`][godoc_logr_logger] exposes [structured logging][site_struct_logging] methods that help create machine-readable logs and adding a wealth of information to log records.
+
+## Default zap logger
+
+Operator SDK uses a `zap`-based `logr` backend when scaffolding new projects. To assist with configuring and using this logger, the SDK includes several helper functions.
+
+In the simple example below, we add the zap flagset to the operator's command line flags with `BindFlags()`, and then set the controller-runtime logger with `zap.Options{}`.
+
+By default, `zap.Options{}` will return a logger that is ready for production use. It uses a JSON encoder, logs starting at the `info` level. To customize the default behavior, users can use the zap flagset and specify flags on the command line. The zap flagset includes the following flags that can be used to configure the logger:
+
+* `--zap-devel`: Development Mode defaults(encoder=consoleEncoder,logLevel=Debug,stackTraceLevel=Warn)
+			  Production Mode defaults(encoder=jsonEncoder,logLevel=Info,stackTraceLevel=Error)
+* `--zap-encoder`: Zap log encoding ('json' or 'console')
+* `--zap-log-level`: Zap Level to configure the verbosity of logging. Can be one of 'debug', 'info', 'error',
+			       or any integer value > 0 which corresponds to custom debug levels of increasing verbosity")
+* `--zap-stacktrace-level`: Zap Level at and above which stacktraces are captured (one of 'warn' or 'error')
+
+Consult the controller-runtime [godocs][logging_godocs] for more detailed flag information.
+
+### A simple example
+
+Operators set the logger for all operator logging in [`cmd/manager/main.go`][code_set_logger]. To illustrate how this works, try out this simple example:
+
+```Go
+package main
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"  
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var globalLog = logf.Log.WithName("global")
+func main() {
+	// Add the zap logger flag set to the CLI. The flag set must
+	// be added before calling flag.Parse().
+	opts := zap.Options{}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+    
+	logger := zap.New(zap.UseFlagOptions(&opts))
+	logf.SetLogger(logger)
+
+	scopedLog := logf.Log.WithName("scoped")
+
+	globalLog.Info("Printing at INFO level")
+	globalLog.V(1).Info("Printing at DEBUG level")
+	scopedLog.Info("Printing at INFO level")
+	scopedLog.V(1).Info("Printing at DEBUG level")
+}
+```
+
+#### Output using the defaults
+```console
+$ go run main.go
+INFO[0000] Running the operator locally in namespace default. 
+{"level":"info","ts":1587741740.407766,"logger":"global","msg":"Printing at INFO level"}
+{"level":"info","ts":1587741740.407855,"logger":"scoped","msg":"Printing at INFO level"}
```

Comment:
Wy here the output does not match with the example?

Reply:
I am not sure I understand, What is not matching?

Reply:
The output msg is not == the values used in the example. Is this output obtained from the code above?

Reply:
![CASE1](https://user-images.githubusercontent.com/56410941/81178351-10439980-8f76-11ea-98bf-803cacaedd6b.png)